android/app/src/main/java/org/uwaterloo/subletr/pages/watcardverification/WatcardVerificationPageView.kt

```diff
@@ -0,0 +1,296 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.components.button.PrimaryButton
+import org.uwaterloo.subletr.components.button.SecondaryButton
+import org.uwaterloo.subletr.theme.SubletrTheme
+import org.uwaterloo.subletr.theme.darkerGrayButtonColor
+import org.uwaterloo.subletr.theme.lightGrayButtonColor
+import org.uwaterloo.subletr.theme.primaryTextColor
+import org.uwaterloo.subletr.theme.secondaryTextColor
+import org.uwaterloo.subletr.theme.subletrBlack
+import org.uwaterloo.subletr.theme.subletrPink
+import org.uwaterloo.subletr.theme.textOnGray
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+
+private const val ELEMENT_WIDTH = 0.75f
+@Composable
+fun WatcardVerificationPageView(
+	modifier : Modifier = Modifier,
+	viewModel : WatcardVerificationPageViewModel = hiltViewModel(),
+	uiState: WatcardVerificationUiState = viewModel.uiStateStream.subscribeAsState(
+		WatcardVerificationUiState.Loading
+	).value
+) {
+
+	if (uiState is WatcardVerificationUiState.Loading) {
+		Column(
+			modifier = modifier
+				.fillMaxSize(1.0f)
+				.imePadding(),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is WatcardVerificationUiState.Loaded) {
+		Column(
+			modifier = modifier.fillMaxSize(1.0f),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			Spacer(modifier = modifier.weight(6.0f))
+			
+			Text(
+				text = stringResource(id = R.string.watcard_verification),
+				style = MaterialTheme.typography.titleMedium,
+				color = primaryTextColor
+			)
+
+			Column(modifier = modifier
```

Comment:
shouldnt need to do this, just have a single text that just flows to the next line